staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -87,8 +87,22 @@ func GetCloudProvider(name string, config io.Reader) (Interface, error) {
 }
 
 // Detects if the string is an external cloud provider
-func IsExternal(name string) bool {
-	return name == externalCloudProvider
+// IsExternal is responsible for serving DisableCloudProviders featureGate and stop cloud.Interface
+// from being initialized in kubelet, kube-controller-manager or kube-api-server
+func IsExternal(name string, cloudProvidersDisabled bool) (bool, error) {
```

Comment:
Prefer not to change the function signature of IsExternal, because it _could_ be consumed externally since k8s.io/cloud-provider is an externally facing package and it could break some clients using it.

Prefer a new function, something like:
```go
func IsInternalCloudProvider(name string) error {
    switch name{
        case "aws", "azure", "gce", "openstack", "vsphere":
          return true
    }
    
    return false
}
```

And from the calling side it could just be:
```go
if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsInternal(s.CloudProvider) {
    klog.Fatalf("....")
} 

if cloudprovider.IsExternal(cloudProvider) { 
    ....
}
```

Reply:
@andrewsykim Implemented this solution. Instead of klog.Fatal I'm returning error in all of the 3 main components initialization phase, so it should be fatal and managed by default logic. PTAL